log/logging.go

```diff
@@ -291,18 +403,36 @@ func (s *Logger) SetTimestampFlags(flags int) {
 	s.l.SetFlags(flags)
 }
 
+// Add a key/value field to every log line from this logger.
+func (s *Logger) SetStaticField(name string, value interface{}) {
+	s.staticArgs[name] = fmt.Sprintf("%v", value)
+}
+
+type logFormatter interface {
```

Comment:
The name of this interface is a bit misleading. It appears to be a formatter but it also has side effects (printing the log) which aren't immediately obvious.

This also strikes me as a good chance to use go's first-class treatment of functions by changing this interface to become a type alias on a function.

Unless we rename this to a `LogPrinter`, I'd definitely leave the actual printing up to whoever uses the formatter (in this case, the logger).

```go
type Formatter func(id string, level LogLevelName /* etc */) string

func formatAsPlainText(id string, level LogLevelName /* etc */) string {
  // ...
}

func formatAsJson(id string, level LogLevelName /* etc */) string {
  // ...
}

type logger struct {
  formatter Formatter
}

func NewJsonLogger() Logger {
  return &logger{formatter: formatAsJson}
}
```

Something along those lines. Playground example: http://play.golang.org/p/z7jjYpcV6a